src/Admin/AdminExtension.php

```diff
@@ -28,19 +27,17 @@ class AdminExtension extends AbstractAdminExtension
     protected $actionManager;
 
     /**
-     * @var TokenStorageInterface|SecurityContextInterface
+     * @var TokenStorageInterface
      */
     protected $securityContext;
 
     /**
-     * NEXT_MAJOR: Go back to signature class check when bumping requirements to SF 2.6+.
-     *
-     * @param TokenStorageInterface|SecurityContextInterface $tokenStorage
+     * @param TokenStorageInterface $tokenStorage
      */
     public function __construct(ActionManagerInterface $actionManager, $tokenStorage)
     {
-        if (!$tokenStorage instanceof TokenStorageInterface && !$tokenStorage instanceof SecurityContextInterface) {
-            throw new \InvalidArgumentException('Argument 2 should be an instance of Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface or Symfony\Component\Security\Core\SecurityContextInterface');
+        if (!$tokenStorage instanceof TokenStorageInterface) {
```

Comment:
Should we replace the if with a signature check as proposed by the comment?

Reply:
I don't know, I didn't change it because of BC, I guess I should have been removed only the `when bumping requirements to SF 2.6+.` part of the comment. 

IMO I don't think someone has extended `AdminExtension`... so I'm ok removing the `if` and add signature check.

Reply:
The `SecurityContextInterface` does not exist in SF 3.4+

Reply:
@franmomu if you're think of signature compatibility issues, it does not apply to the constructor (neither does the Liskov Substitution Principle).

Reply:
I'll change it, hadn't thought about that.